instagram/src/components/PostContainer/PostsPage.js

```diff
@@ -0,0 +1,176 @@
+import dummyData from "../../dummy-data";
+import SearchBar from "../SearchBar/SearchBar";
+import PostContainer from "./PostContainer";
+import "../../css/main.css";
+
+class PostsPage extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      data: []
+    };
+  }
+
+  componentDidMount() {
+    // Retrieves stringified state object from localStorage (if any):
+    const retrievedState = localStorage.getItem("storedState");
+    // Parses state object back to its original formatting:
+    const parsedState = JSON.parse(retrievedState);
+    // Uses state object from localStorage if there is one. If not, uses dummyData as before:
+    if (parsedState) {
+      this.setState({
+        data: parsedState.data,
+        newCommentText: parsedState.newCommentText,
+        searchText: parsedState.searchText
+      });
+    } else if (!parsedState) {
+      // There might be a more succinct way to do this, but the following turns the timestamp string from dummyData into Unix Epoch Time (while leaving the other data alone). Earlier I had just edited dummy-data.js, so this is more elegant.
+      const data = dummyData.map(post => {
+        return (post = {
+          username: post.username,
+          visible: true,
+          thumbnailUrl: post.thumbnailUrl,
+          imageUrl: post.imageUrl,
+          likes: post.likes,
+          liked: false,
+          timestamp: moment(
+            post.timestamp,
+            "MMMM Do YYYY, h:mm:ss a"
+          ).valueOf(),
+          comments: post.comments,
+          newCommentText: ""
+        });
+      });
+      this.setState({
+        data: data,
+        searchText: ""
+      });
+    }
+  }
+
+  // Note: I think I have to call prevProps in order to call prevState, even though I don't use prevProps:
+  componentDidUpdate(prevProps, prevState) {
+    // Copies state into localStorage each time App.js is updated:
+    let storedState = this.state;
+    localStorage.setItem("storedState", JSON.stringify(storedState));
+    // I put the search function inside CDU so I could be sure that this.state.searchText is current before setting the visible property accordingly. Required conditional to avoid an infinite loop since I have CDU calling setState():
+    if (this.state.searchText !== prevState.searchText) {
+      // Once again, to get at the data nested in this.state, I need a spread operator:
+      let tempData = [...this.state.data];
+      // It was nice to use the humble forEach to solve a problem. Here forEach() checks whether post.username includes the current searchText, and updates post.visible as true or false accordingly:
+      tempData.forEach(post => {
+        if (!post.username.includes(this.state.searchText)) {
+          post.visible = false;
+        } else if (post.username.includes(this.state.searchText)) {
+          post.visible = true;
+        }
+      });
+      this.setState({
+        data: tempData
+      });
+    }
+  }
```

Comment:
very nice way to store the current state within storage -- for something a little more intuitive that saves you some time writing the code, take a look at the library `react-simple-storage` and how it can be utilized to do some of the heavy lifting for you